researchstack-sdk/src/main/java/org/sagebionetworks/bridge/researchstack/BridgeDataProvider.java

```diff
@@ -167,63 +181,90 @@ public boolean isConsented() {
     // TODO: getConsent rid of Consent methods below on the interface. let ConsentManager handle the
     // implementation details and expose leave giveConsent, getConsent, withdrawConsent, and
     // isConsented
+    @Nullable
     @Override
     public ConsentSignatureBody loadLocalConsent(Context context) {
-
-        return createConsentSignatureBody(authenticationManager.getConsentSync(bridgeConfig.getStudyId()));
+        ConsentSignatureBody consent = createConsentSignatureBody(
+                authenticationManager.getConsentSync(bridgeConfig.getStudyId()));
+        logger.debug("loadLocalConsent called, got: ");
+        return consent;
     }
 
     @Override
-    public void saveConsent(Context context, TaskResult consentResult) {
-        giveConsent(bridgeConfig.getStudyId(), createConsentSignature(consentResult));
+    public void saveConsent(Context context, @NonNull TaskResult consentResult) {
+        logger.warn("saveConsent called -- DEPRECATED");
+        giveConsent(bridgeConfig.getStudyId(), createConsentSignature(consentResult))
+                .toBlocking().value();
     }
 
     @Override
     public void saveLocalConsent(Context context, ConsentSignatureBody signatureBody) {
-        giveConsent(bridgeConfig.getStudyId(), createConsentSignature(signatureBody));
+        logger.debug("saveLocalConsent called for: " + signatureBody);
+        LocalDate birthdate = null;
+        if (signatureBody.birthdate != null) {
+            birthdate = LocalDate.fromDateFields(signatureBody.birthdate);
+        }
```

Comment:
Would it be useful to have a wrapper function that takes in a ConsentSignature and calls authenticationManager.giveConsentSync()? This way, you can leverage the null checks and type conversions of createConsentSignature() and avoid writing these checks in two different placse.